### DISCUSSION TITLE
No way to build ios preview without paid account?

### DISCUSSION THREAD

**Discussion by vedmant** [expo/expo#27489] [Archived: Questions] [answered]
Hi, I want to preview and test my app on iOs but I don't have paid apple account and looks like expo doesn't allow to do so, however it's possible to build and run iOs app with react native without paid account. Any way to still do this with expo or only way is to ditch expo and work with clean react-native install?

**Comment by tsapeta** [comment]
You can also do it with Expo. Just run `npx expo prebuild` that will generate `ios` folder for you, then you can use it just like any other React Native project. There is no need to use EAS in this case.

**Reply by vedmant** [reply]
@tsapeta I was trying to do so, but having errors all the time, for example when I run `react-native run-ios --mode Release` I have: 
"error Cannot start server in new window because no terminal app was specified.".

**Reply by vedmant** [reply]
Update, looks like I made it, will post instructions for anyone interested.

**Comment by vedmant** [comment, marked as answer]
Ok, looks like I figured out how to do this, so following steps are needed:

1. `npx expo prebuild`
2. `open ios/YourProject.xcworkspace`
3.  Go to project settings "Signing & Capabilities" and select your team
4. Go to project settings "Signing & Capabilities" and remove "Push Notifications"
5. Click on top bar your project name, select "Edit Scheme", select "Build Configuration" - Release
6. Build bundle with `npx expo export:embed --entry-file='node_modules/expo/AppEntry.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'`
7. In Xcode in Build phases -> Bundle React Native code and images -> Check off "For install builds only" if it's checked
8. Go to Xcode, select your connected device and run the project

I did many iterations, but this looks like fixed the problem for me. 
Now I can test my app on my iPhone in release mode without need to connect to dev server.

I see still that `npx eact-native run-ios --mode Release --terminal terminal`` not working for some reason, fails to build.

**Reply by artabon** [reply]
Did you misspell react in 'npx **eact-native** run-ios --mode Release --terminal terminal`'

**Reply by saru2020** [reply]
6th step fails for me with 
```
Using src/app as the root directory for Expo Router.
Starting Metro Bundler
iOS node_modules/expo/AppEntry.js ░░░░░░░░░░░░░░░░  0.0% (1/1)Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 

None of these files exist:
  * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
  * App
  1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
  2 |
> 3 | import App from '../../App';
    |                  ^
  4 |
  5 | registerRootComponent(App);
  6 |
Error: Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 

None of these files exist:
  * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
  * App
  1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
  2 |
> 3 | import App from '../../App';
    |                  ^
  4 |
  5 | registerRootComponent(App);
  6 |
Error: Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 

None of these files exist:
  * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
  * App
  1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
  2 |
> 3 | import App from '../../App';
    |                  ^
  4 |
  5 | registerRootComponent(App);
  6 |
    at ModuleResolver.resolveDependency (/usr/local/lib/node_modules/expo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:112:15)
    at DependencyGraph.resolveDependency (/usr/local/lib/node_modules/expo/node_modules/metro/src/node-haste/DependencyGraph.js:236:43)
    at /usr/local/lib/node_modules/expo/node_modules/metro/src/lib/transformHelpers.js:165:21
    at resolveDependencies (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:42:25)
    at visit (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:83:30)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0)
    at buildSubgraph (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:103:3)
    at Graph._buildDelta (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/Graph.js:157:22)
    at Graph.initialTraverseDependencies (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/Graph.js:140:19)
```

is this approach still working?

**Reply by YasserDbeis** [reply]
> 6th step fails for me with 
> 
> ```
> 
> Using src/app as the root directory for Expo Router.
> 
> Starting Metro Bundler
> 
> iOS node_modules/expo/AppEntry.js ░░░░░░░░░░░░░░░░  0.0% (1/1)Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 
> 
> 
> 
> None of these files exist:
> 
>   * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
> 
>   * App
> 
>   1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
> 
>   2 |
> 
> > 3 | import App from '../../App';
> 
>     |                  ^
> 
>   4 |
> 
>   5 | registerRootComponent(App);
> 
>   6 |
> 
> Error: Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 
> 
> 
> 
> None of these files exist:
> 
>   * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
> 
>   * App
> 
>   1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
> 
>   2 |
> 
> > 3 | import App from '../../App';
> 
>     |                  ^
> 
>   4 |
> 
>   5 | registerRootComponent(App);
> 
>   6 |
> 
> Error: Unable to resolve module ../../App from /Users/saravanan/Documents/personal/github/GoalsTracker_MobileApp/GoalsTracker/node_modules/expo/AppEntry.js: 
> 
> 
> 
> None of these files exist:
> 
>   * App(.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.mjs|.native.mjs|.mjs|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.cjs|.native.cjs|.cjs|.ios.scss|.native.scss|.scss|.ios.sass|.native.sass|.sass|.ios.css|.native.css|.css|.ios.svg|.native.svg|.svg)
> 
>   * App
> 
>   1 | import registerRootComponent from 'expo/src/launch/registerRootComponent';
> 
>   2 |
> 
> > 3 | import App from '../../App';
> 
>     |                  ^
> 
>   4 |
> 
>   5 | registerRootComponent(App);
> 
>   6 |
> 
>     at ModuleResolver.resolveDependency (/usr/local/lib/node_modules/expo/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:112:15)
> 
>     at DependencyGraph.resolveDependency (/usr/local/lib/node_modules/expo/node_modules/metro/src/node-haste/DependencyGraph.js:236:43)
> 
>     at /usr/local/lib/node_modules/expo/node_modules/metro/src/lib/transformHelpers.js:165:21
> 
>     at resolveDependencies (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:42:25)
> 
>     at visit (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:83:30)
> 
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
> 
>     at async Promise.all (index 0)
> 
>     at buildSubgraph (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/buildSubgraph.js:103:3)
> 
>     at Graph._buildDelta (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/Graph.js:157:22)
> 
>     at Graph.initialTraverseDependencies (/usr/local/lib/node_modules/expo/node_modules/metro/src/DeltaBundler/Graph.js:140:19)
> 
> ```
> 
> 
> 
> is this approach still working?

You need to change the entry point path to your own code's entry point.

**Reply by saru2020** [reply]
> --entry-file='node_modules/expo/AppEntry.js' --bundle-output='./ios/main.jsbundle'

I don't see either of these files, that's what the command is complaining about, I even tried regenerating the ios/android folders, what else am I missing?

**Reply by takeru** [reply]
in package.json

> "main": "expo-router/entry"

> ls node_modules/expo-router/entry.js

> npx expo export:embed --entry-file='node_modules/expo-router/entry.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'

**Reply by Gap6769** [reply]
Is there anyway to actually get the .ipa from this method to sideload it?

**Reply by tunnaduong** [reply]
@Gap6769 yes, there is. https://www.youtube.com/watch?v=ME4_RHCaCAk

**Reply by ShingLi** [reply]
> Ok, looks like I figured out how to do this, so following steps are needed:
> 
> 1. `npx expo prebuild`
> 2. `open ios/YourProject.xcworkspace`
> 3. Go to project settings "Signing & Capabilities" and select your team
> 4. Go to project settings "Signing & Capabilities" and remove "Push Notifications"
> 5. Click on top bar your project name, select "Edit Scheme", select "Build Configuration" - Release
> 6. Build bundle with `npx expo export:embed --entry-file='node_modules/expo/AppEntry.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'`
> 7. In Xcode in Build phases -> Bundle React Native code and images -> Check off "For install builds only" if it's checked
> 8. Go to Xcode, select your connected device and run the project
> 
> I did many iterations, but this looks like fixed the problem for me. Now I can test my app on my iPhone in release mode without need to connect to dev server.
> 
> I see still that `npx eact-native run-ios --mode Release --terminal terminal`` not working for some reason, fails to build.

Hello, I'm currently running into an issue. Since I don't have a Mac, I can only use EAS cloud builds. However, it seems to always get stuck at the Apple paid developer account step. Is there a way to just use EAS for the build, and then use other methods to sign the IPA and install it on my iPhone?

**Comment by YasserDbeis** [comment]
Apparently, EXPO_USE_METRO_WORKSPACE_ROOT was enabled by default starting with Expo SDK 52. So if you are using expo 52 and above, and are working in a monorepo, you might get an error saying it cant find the entry point because it doubles up part of the path to the entry point file. In my case, I was using the tamagui takout monorepo set up, and kept getting the error: `error: The resource `/path-to-project-root/**apps/expo/apps/expo**/index.js` was not found`.

To fix this, I found that adding `export EXPO_NO_METRO_WORKSPACE_ROOT=1` to the top of my `.xcode.env.local` file did the trick.
Hope this helps!

**Reply by YasserDbeis** [reply]
Actually, as it turns out, this was actually due to a conflcit with expo-updates dependency in my project. Not sure the relation but if you do not have expo-updates this should not be an issue!
